include/substrait-mlir/Dialect/Substrait/IR/SubstraitOps.td

```diff
@@ -539,24 +539,22 @@ def SetOpKind : I32EnumAttr<"SetOpKind",
 def Substrait_SetOp : Substrait_RelOp<"set", [
     DeclareOpInterfaceMethods<InferTypeOpInterface>
   ]> {
-  // TODO(daliashaaban): This will evolve into an operation that supports more than two inputs.
   let summary = "set operation";
   let description = [{
     Represents the `SetRel` message together with the `RelCommon`, input
     `Rel`, and `SetOp` enumeration it contains. This operation performs 
     relational set operations such as intersection, union, and difference.
-    The current implementation only allows for two inputs (instead of two or more).
+    There must be at least two inputs.
   }];
   
   let arguments = (ins
-    Substrait_Relation:$left,
-    Substrait_Relation:$right,
+    Variadic<Substrait_Relation>:$inputs,
```

Comment:
You do this test in `inferReturnTypes` and you may *have* to do it there (depending on the order of the verifiers and the type inference, which I don't know by heart) but I would *also* add a constraint on `inputs` here that says that there must be at least two inputs. I think there should be something in `OpBase.td` that does that. Having it here shows it in the typical first place you look up the definition of an op, so at the very least it's documentation.